<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Parameters -->
    <arg name="use_sim_time" default="false"/>
    <param name="use_sim_time" value="$(arg use_sim_time)"/>

    <!-- Input Topics -->
    <arg name="imu_topic_name" default="/gt_box/ap20/imu"/>
    <arg name="prism_position_topic_name" default="/gt_box/ap20/prism_position"/>
    <arg name="gnss_position_topic_name" default="/gt_box/cpt7/gps/fix"/>
    <arg name="gnss_offline_pose_topic_name" default="/gt_box/inertial_explorer/odometry"/>

    <!-- Config Files -->
    <arg name="core_graph_config_file" default="$(find atn_position3_fuser)/config/core/core_graph_config.yaml"/>
    <arg name="core_graph_parameter_file" default="$(find atn_position3_fuser)/config/core/core_graph_params.yaml"/>
    <arg name="core_extrinsic_file" default="$(find atn_position3_fuser)/config/core/core_extrinsic_params.yaml"/>
    <arg name="position_graph_parameter_file"
         default="$(find atn_position3_fuser)/config/position_graph_specific/position_graph_graph_params.yaml"/>
    <arg name="position_extrinsic_file"
         default="$(find atn_position3_fuser)/config/position_graph_specific/position_graph_extrinsic_params.yaml"/>
    <arg name="trajectory_alignment_parameter_file"
         default="$(find atn_position3_fuser)/config/position_graph_specific/trajectory_alignment_params.yaml"/>

    <!-- Logging Location -->
    <arg name="logging_dir_location" default="$(find atn_position3_fuser)/logging"/>

    <!-- Node -->
    <node pkg="atn_position3_fuser" type="atn_position3_fuser_node" name="atn_position3_fuser_node"
          output="screen"> <!-- launch-prefix="gdb -args"-->
        <!-- Launch Parameters -->
        <param name="launch/optimizationResultLoggingPath" type="str" value="$(arg logging_dir_location)"/>
        <!-- Parameter files -->
        <rosparam command="load" file="$(arg core_graph_config_file)"/>
        <rosparam command="load" file="$(arg core_graph_parameter_file)"/>
        <rosparam command="load" file="$(arg core_extrinsic_file)"/>
        <rosparam command="load" file="$(arg position_graph_parameter_file)"/>
        <rosparam command="load" file="$(arg position_extrinsic_file)"/>
        <rosparam command="load" file="$(arg trajectory_alignment_parameter_file)"/>
        <!-- Remapping of topics -->
        <remap from="/imu_topic" to="$(arg imu_topic_name)"/>
        <remap from="/prism_position_topic" to="$(arg prism_position_topic_name)"/>
        <remap from="/gnss_position_topic" to="$(arg gnss_position_topic_name)"/>
        <remap from="/gnss_offline_pose_topic" to="$(arg gnss_offline_pose_topic_name)"/>
    </node>

</launch>
